Update Semgrep CI (#472)
returntocorp/semgrep-docs: docs/semgrep-ci/overview.md

Semgrep CI runs fully in your build environment: **your code is never sent anywhere**.
:::

> **Note**: Semgrep CI is a subcommand of Semgrep CLI, so you can test the code and behavior of Semgrep CI in your command line using `semgrep ci` command. This means that all the capabilities of Semgrep CLI are available in Semgrep CI.

Author: Tried to summarize the note as you suggested. Not sure if the first sentence is entirely true (about testing it). Feel free to suggest anything you would like to see here, please. This should be a friendly introduction to our CI, it does not have to be very technical, it should just give an image of possibilities. @brendongo @underyx

Reviewer: ```suggestion
**Note**: Semgrep CI is a subcommand of Semgrep CLI, so you can test the code and behavior of Semgrep CI on your machine by running `semgrep ci`.
```
I tried to remove some mentions of the word 'command'. I also removed the 'capabilities of Semgrep CLI' sentence which sounded very vague to me.

### Other CI providers

The community successfully run Semgrep CI on the following platforms (some include a link to sample configuration):

- Bitbucket Pipelines
- Bitrise
- Buildbot
- Buildkite [(sample configuration)](../sample-ci-configs/#buildkite)
- CircleCI [(sample configuration)](../sample-ci-configs/#circleci) 
- Codefresh
- Jenkins [(sample configuration)](../sample-ci-configs/#jenkins)
- TeamCity CI
- Travis CI

To add Semgrep CI to any CI environment, use the [`returntocorp/semgrep`](https://hub.docker.com/r/returntocorp/semgrep) Docker image directly.

For full project scans:

```sh

> docker run -v $(pwd):/src --workdir /src returntocorp/semgrep semgrep ci --config auto --config <other rule or rulesets>

Author: Is this still valid? @brendongo @underyx

Reviewer 2: Yes but `semgrep ci --config auto` also works (without using docker) and is more equivalent to the line you are replacing

Author: ![Screenshot 2022-04-08 at 12 10 02](https://user-images.githubusercontent.com/26669307/162415115-c3e0494d-55cb-462e-b7d2-660ab82a4aba.jpg)
There is no line being replaced actually... I just deleted a baseline mentioned above, because we explain it in a different article. I have not touched this command...

Author: @brendongo why somebody would like to use such a long command? Maybe this comes from some older version of the agent?

Reviewer 2: It was because you could not pip install semgrep-agent so had to use docker

Author: @brendongo @underyx Substituted with suggested text. Please check it. This is under a "Other CI providers" section in the Semgrep CI overview document.

```
For full project scans:

```sh
semgrep ci --config auto --config <other rule or rulesets>
```

**Note**: If you need to use a different image than docker, install Semgrep CI by `pip install semgrep`.
```

To connect your Semgrep CI scans to Semgrep App, you can optionally provide the following environment variables:

#### Sample configurations

These instructions have been used on the following providers by the community:

- Bitbucket Pipelines TODO ask why there are no links
- Bitrise
- Buildbot
- TeamCity CI
- Codefresh
- Travis CI
- Buildkite [(sample configuration)](../sample-ci-configs/#buildkite)
- Jenkins [(sample configuration)](../sample-ci-configs/#jenkins)

> - CircleCI [(sample configuration)](../sample-ci-configs/#circleci) 

Author: @pabloest maybe you can help here. 🙏 

There used to be [this table](https://semgrep.dev/docs/semgrep-ci/overview/#run-semgrep_agent) with other CI providers. It included GitHub and GitLab as well, but we already link to these samples of configuration in the sections dedicated to the above. I guess that the document evolved organically and this table might have been one of the older foundations.
 
I made a list from this table as we do not have to make a table with two columns for a list essentially (there are no values different in nature from the first column in the second one).

Action: Please, do we need to have CI providers such as Codefresh or Bitrise listed here if we do not provide sample CI configs for them? If yes, who might provide us with these sample configs for Semgrep CI (we were checking this together with @brendongo so he was not sure as well). And sorry, they are not sorted out alphabetically, I may sort them out.

Reviewer 3: According to the docs it seems at some point one or more users reported success in running Semgrep on those CI platforms. However I haven’t been able to find evidence of that for Codefresh or Bitrise. But I have seen it for Travis CI (for example). So I would say where we can see evidence of success in a CI platform, let’s list it even if we don’t provide a sample config (that would be even better, of course).

Author: That is nice @pabloest, adding as the following text:

`The community succesfully run Semgrep CI on the following platforms (some include a link to sample configuration):

- Bitbucket Pipelines TODO ask why there are no links
- Bitrise
- Buildbot
- Buildkite [(sample configuration)](../sample-ci-configs/#buildkite)
- CircleCI [(sample configuration)](../sample-ci-configs/#circleci) 
- Codefresh
- Jenkins [(sample configuration)](../sample-ci-configs/#jenkins)
- TeamCity CI
- Travis CI`

Reviewer 3: Thanks!

I suggest this edit: “…successfully ran Semgrep in the following CI platforms…”

Author: @pabloest oh thank you, forgot to use past tense. 🤦
